Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -685,11 +690,12 @@ namespace Terrain
         meshHeights.resize_no_construct(outputSamplesCount);
         meshNormals.resize_no_construct(outputSamplesCount);
 
-        auto perPositionCallback = [this, &heights, querySamplesX]
-        (size_t xIndex, size_t yIndex, const AzFramework::SurfaceData::SurfacePoint& surfacePoint, [[maybe_unused]] bool terrainExists)
+        auto perPositionCallback = [this, &heights, querySamplesX, &terrainExistsAnywhere]
+        (size_t xIndex, size_t yIndex, const AzFramework::SurfaceData::SurfacePoint& surfacePoint, bool terrainExists)
         {
             const float height = surfacePoint.m_position.GetZ() - m_worldBounds.GetMin().GetZ();
-            heights.at(yIndex * querySamplesX + xIndex) = height;
+            heights.at(yIndex * querySamplesX + xIndex) = terrainExists ? height : -1.0f; // store terrain doesn't exist as a -1.0
```

Comment:
Nit: The "-1.0f" should probably be a named constant. (Ex: "NoTerrainHeight" or something)

Reply:
this is probably ok to do, but just to be clear there's nothing special about -1.0, other than the fact that it's negative. All subsequent checks aren't for that specific value, they're just checking for less than zero.